### ISSUE TITLE
Not an issue, just a question about your dev branch

### ISSUE THREAD

**Issue by pkdavidson** [khibma/ArcGISAdministrationToolkit#13]
Hi Kevin;
I noticed the devupdates branch has a number of  updates, such as removing of Requests, better https support, etc...
Is that branch usable as a beta version? 
A quick perusal showed some comments about some code being 2.7 compatible only.
That's fine for me, although I can imagine you want full Py3 support.

Any idea when you plan on integrating your dev branch into the master branch?

Great stuff, thanks for the hard work.

**Comment by pkdavidson** [comment]
I am guessing that the master branch is not https compatible? 

I am using the master branch code since it's released but it seems like I tried this also with the dev branch and had much the same results.  

Executing: ReportServerInfo FQDN 6443 admin pword
Start Time: Fri Oct 07 18:07:11 2016
Running script ReportServerInfo...
Failed script ReportServerInfo...

Traceback (most recent call last):
  File "C:\arcgisserver\ServerAdminToolkit\ReportServerInfo.py", line 120, in <module>
    getServerInfo(server, port, adminUser, adminPass)
  File "C:\arcgisserver\ServerAdminToolkit\ReportServerInfo.py", line 44, in getServerInfo
    token = gentoken(server, port, adminUser, adminPass)
  File "C:\arcgisserver\ServerAdminToolkit\ReportServerInfo.py", line 26, in gentoken
    token = json.loads(urllib.urlopen(url + "?f=json", query_string).read())
  File "C:\Python27\ArcGIS10.2\Lib\json\__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "C:\Python27\ArcGIS10.2\Lib\json\decoder.py", line 365, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Python27\ArcGIS10.2\Lib\json\decoder.py", line 383, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded

Failed to execute (ReportServerInfo).
Failed at Fri Oct 07 18:07:11 2016 (Elapsed Time: 0.01 seconds)

When I run the same script against port 6080, it runs fine.
The Server is setup for both HTTP and HTTPS


Thanks !
(fyi... out of town next week, I'm not ignoring this if I don't respond.)

**Comment by khibma** [comment]
hi @pkdavidson thanks for poking at the dev branch. You could certainly consider it a "beta".
The entire dev branch should be py2/py3 compatible.. I'll have to search out what isnt, because it should be.

The error you're reporting - its in getting a token. I changed the code to ask the server if HTTPS is enabled, if it is, all requests are then made over HTTPS. If the script cant get this info, it uses HTTP and this gets called `print("Error getting SSL setting, using HTTP")` (I'd expect you'd see that message). Further on this point your specific error is happening because what came back from the token wasn't something I could put into JSON. (I should better catch that error) This generally happens when the request was bad and the server sends back HTML instead of JSON.
On line 125: `        return json.loads(response_text)` you could change that to

```
try:
        return json.loads(response_text)`
except:
        print("failed to get good response. Response is: \n {}".format(response_text))
        sys.exit()
```

Could you tell me your parameters to running the script? Not your user/pass, but the server (like is it just server or server.domain.com) and what port you're using?
Also what is the status of SSL on your server? enabled to just SSL, or is SSL-No ssl enabled?


(I must not have notifications turned on .... I never see when an issue is added )

**Comment by khibma** [comment]
Merged PR #11 which should address SSL problems.
